src/rust/engine/process_execution/src/lib.rs

```diff
@@ -177,6 +178,63 @@ fn serialize_level<S: serde::Serializer>(level: &log::Level, s: S) -> Result<S::
   s.serialize_str(&level.to_string())
 }
 
+///
+/// Input Digests for a process execution. The `complete` Digest is the computed union of all
+/// inputs: the rest of the Digests should be disjoint (or have identical contents where they
+/// overlap).
+///
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize)]
+pub struct InputDigests {
+  ///
+  /// All of the input Digests, merged.
+  ///
+  pub complete: Digest,
+
+  ///
+  /// The input files for the process execution, which will be materialize as mutable inputs in a
+  /// sandbox for the process.
+  ///
+  pub input_files: Digest,
+
+  ///
+  /// If non-empty, the Digest of a nailgun server to use to attempt to spawn the Process.
+  ///
+  pub use_nailgun: Digest,
```

Comment:
Should this just be called `tool_files` or `tool_digest`? Or would such a renaming be better done in its own PR?

Reply:
(This also relates to the "reusable input digests" PR, maybe we should just call these kinds of digests "tool" since they are associated with the "tool" being run.

Reply:
> Should this just be called `tool_files` or `tool_digest`? Or would such a renaming be better done in its own PR?

Maybe? But right now this being non-`EMPTY_DIGEST` is also treated as the trigger to attempt to spawn as `nailgun`, and I have been assuming that we would have a separate field for the `immutable_input_*`s, which don't need to differentiate between being for the tool or not. #12716 will be coming soon regardless, and can change the naming if that ends up looking right.

I think that Bazel calling them "tool_*" has to do with the fact that the worker protocol is not JVM specific (...neither is nailgun, really. but in practice).